packages/react/src/components/Tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,159 @@
+import type { HTMLAttributes } from 'react';
+import React, { cloneElement, forwardRef, useMemo, useState } from 'react';
+import cn from 'classnames';
+import {
+  useFloating,
+  autoUpdate,
+  offset,
+  flip,
+  shift,
+  arrow,
+  useHover,
+  useFocus,
+  useDismiss,
+  useRole,
+  useInteractions,
+  useTransitionStyles,
+  FloatingArrow,
+  FloatingPortal,
+} from '@floating-ui/react';
+
+import styles from './Tooltip.module.css';
+import mergeRefs from './utils';
+
+const ARROW_HEIGHT = 7;
+const GAP = 4;
+
+export type TooltipProps = {
+  /**
+   * The element that triggers the tooltip.
+   * @note Needs to be a single ReactElement and not: <React.Fragment/> | <></>
+   */
+  children: React.ReactElement & React.RefAttributes<HTMLElement>;
+  content: string;
+  /**
+   * Placement of the tooltip on the trigger.
+   * @default 'top'
+   */
+  placement?: 'top' | 'right' | 'bottom' | 'left';
+  /** Delay in milliseconds before opening.
+   * @default 150
+   */
+  delay?: number;
+  open?: boolean;
+  defaultOpen?: boolean;
+} & HTMLAttributes<HTMLDivElement>;
+
+export const Tooltip = forwardRef<HTMLDivElement, TooltipProps>(
+  (
+    {
+      children,
+      content,
+      placement = 'top',
+      delay = 150,
+      open: userOpen,
+      defaultOpen = false,
+      className,
+      ...restHTMLProps
+    },
+    ref,
+  ) => {
+    const [isOpen, setIsOpen] = useState(defaultOpen);
+
+    const arrowRef = React.useRef<SVGSVGElement>(null);
+
+    const { refs, floatingStyles, context } = useFloating({
+      open: userOpen ?? isOpen,
```

Comment:
Maybe make a `const` out of this (`userOpen ?? isOpen`) since we are using it several places. Call it something like `shouldOpen`?